pkg/destroy/aws.go

```diff
@@ -1,22 +1,32 @@
 package destroy
 
 import (
-	"github.com/openshift/installer/pkg/destroy/aws"
+	"github.com/aws/aws-sdk-go/aws"
+	session "github.com/openshift/installer/pkg/asset/installconfig/aws"
+	destroy "github.com/openshift/installer/pkg/destroy/aws"
 	"github.com/openshift/installer/pkg/types"
 	"github.com/sirupsen/logrus"
 )
 
 // NewAWS returns an AWS destroyer from ClusterMetadata.
 func NewAWS(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (Destroyer, error) {
-	filters := make([]aws.Filter, 0, len(metadata.ClusterPlatformMetadata.AWS.Identifier))
+	filters := make([]destroy.Filter, 0, len(metadata.ClusterPlatformMetadata.AWS.Identifier))
 	for _, filter := range metadata.ClusterPlatformMetadata.AWS.Identifier {
 		filters = append(filters, filter)
 	}
 
-	return &aws.ClusterUninstaller{
+	awsSession, err := session.GetSession()
+	if err != nil {
+		return nil, err
+	}
+
+	return &destroy.ClusterUninstaller{
 		Filters: filters,
 		Region:  metadata.ClusterPlatformMetadata.AWS.Region,
 		Logger:  logger,
+		Session: awsSession.Copy(&aws.Config{
```

Comment:
Don't we want to do this session copy inside the `ClusterUninstaller` function to ensure that we are always using the right region even when the function is supplied a session from elsewhere?

Reply:
> Don't we want to do this session copy inside the `ClusterUninstaller` function...

Done with 72319c7f4 -> bb4396613.